src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -25,13 +26,14 @@ public async Task Start(OnMessage onMessage, OnError onError)
             Logger.DebugFormat("Receiver {0} is starting.", receiver.Id);
 
 
-            await receiver.Initialize(pushRuntimeSettings, onMessage, onError).ConfigureAwait(false);
-            await receiver.StartReceive().ConfigureAwait(false);
+            await receiver.Initialize(pushRuntimeSettings, onMessage, onError, cancellationToken).ConfigureAwait(false);
+            await receiver.StartReceive(cancellationToken).ConfigureAwait(false);
 
             isStarted = true;
         }
 
-        public async Task Stop()
+        // TODO: Not sure about using a token here...
```

Comment:
Since its task returning and called when we are shutting down passing the stop token makes sense to me?